some warnings should be removed

https://ci.appveyor.com/project/gmaurel/uncrustify/builds/20205577/job/b7108t3ju45fo5cp
says:
```.txt
  option_enum.cpp
c:\projects\uncrustify\build\src\option_enum.cpp(401): warning C4809: switch statement has redundant 'default' label; all possible 'case' labels are given [C:\projects\uncrustify\build\uncrustify.vcxproj]
```

https://travis-ci.org/uncrustify/uncrustify/jobs/453466109
says:
```.txt
src/option_enum.cpp:20:32: warning: unknown option after ‘#pragma GCC diagnostic’ kind [-Wpragmas]
 #pragma GCC diagnostic ignored "-Wswitch-bool"
```

Any comments, bugfix?

--- a/src/option_enum.cpp.in
+++ b/src/option_enum.cpp.in
@@ -10,7 +10,9 @@
 #include <strings.h>  // strcasecmp()
 #endif
 
-#ifdef __GNUC__
+#if defined(_MSC_VER)
+#pragma warning(disable: 4809)
+#elif defined(__GNUC__)
 #pragma GCC diagnostic ignored "-Wswitch-bool"
 #endif
 

